### PR TITLE
fix(jvm): Eliminate the need to use '--add-exports' VM option

### PR DESCRIPTION
**Pull request by the-sheev**
Tested on Java 11 and Java 16.

### CODE REVIEW

**Review by bluestreak01 — CHANGES_REQUESTED** [review]
perhaps we should delete windows dockerfile to avoid confusion

File: core/Dockerfile-windows [hunk 1]
```diff
@@ -6,34 +6,29 @@ RUN curl -L "https://github.com/git-for-windows/git/releases/download/v2.25.0.wi
...
 RUN c:\apache-maven-3.6.3\bin\mvn clean package -DskipTests -P build-web-console,build-binaries,use-built-in-nodejs
 
-## unpack the runtime
-
-WORKDIR c:/questdb/core/target
-RUN tar xvfz questdb-5.0.6.1-rt-windows-amd64.tar.gz
-#RUN dir questdb-*-rt-windows-amd64.tar.gz
+FROM winamd64/openjdk:16-nanoserver-1809
 
-FROM mcr.microsoft.com/windows/nanoserver:1903
```
[the remark this answers is not in the document]
Comment (the-sheev) [reply core/Dockerfile-windows:29]: It still creates the image without build artifacts. I replaced it with the line above.

File: win64svc/src/questdb.c [hunk 2]
```diff
@@ -83,7 +83,6 @@ void buildJavaArgs(CONFIG *config) {
     LPCSTR javaOpts = "-XX:+UnlockExperimentalVMOptions"
                       " -XX:+AlwaysPreTouch"
                       " -XX:+UseParallelOldGC"
```
Comment (bluestreak01) [inline win64svc/src/questdb.c:85]: This flag does not work on JDK 16:

image

File: win64svc/src/questdb.c [hunk 3]
```diff
@@ -83,7 +83,6 @@ void buildJavaArgs(CONFIG *config) {
     LPCSTR javaOpts = "-XX:+UnlockExperimentalVMOptions"
```
Comment (bluestreak01) [inline win64svc/src/questdb.c:83]: we need to compile the executable, it is checked into `git` repo

File: core/Dockerfile [hunk 4]
```diff
@@ -13,17 +13,11 @@ WORKDIR /build/questdb
 
 RUN ../apache-maven-3.6.3/bin/mvn clean package -DskipTests -P build-web-console,build-binaries,use-built-in-nodejs
 
-WORKDIR /build/questdb/core/target
+FROM openjdk:16-alpine
```
Comment (bluestreak01) [inline core/Dockerfile:16]: using generic image overinflates the image size:

image

File: core/Dockerfile [hunk 5]
```diff
@@ -42,5 +36,7 @@ EXPOSE 9000/tcp
 EXPOSE 8812/tcp
 EXPOSE 9009/tcp
 
+WORKDIR /app
+
 # Run questdb when the container launches
-CMD ["/app/bin/java", "--add-exports", "java.base/jdk.internal.math=io.questdb", "-m", "io.questdb/io.questdb.ServerMain", "-d", "/root/.questdb", "-f"]
```
Comment (bluestreak01) [inline core/Dockerfile:46]: we need to use "RT" binary to keep image size small

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

**Comment by the-sheev** [comment]
I used `--release` to cross-compile with Maven to use Java 16 to run `jlink` (otherwise, compilation against JDK 16 fails). Unfortunately, `--release` does not work with `--add-exports`, so we will need to eliminate usages of `FDBigInteger` before we can merge this pull request (related: #862).

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**